tests/optimization/test_delay_expressions.py

```diff
@@ -75,6 +77,26 @@ def test_no_history(self):
                     "Extrapolating t0 value backwards in time.",
                 ],
             )
+        results = problem.extract_results()
+
+        # Test inline mode
+        problem_inline = ModelNoHistory(inline_delay_expressions=True)
```

Comment:
Test succeeds with `inline_delay_expressions=False` as well

Reply:
I'd say to check the _pymoca_delay value in the results, based on the current approach. It should be zero for `inline_delay_expressions=True` right? And non-zero for `inline_delay_expressions=False`

Reply:
I feel that approach would be exposing internals (in the form of private variables) to the tests, which I'm not convinced is a very clean approach?

Reply:
I'm not sure I follow. The whole idea of this PR is to change the behavior of those variables no? Then we should check that we do it correctly. 

Their names being prefixed with an underscore is more to indicate "don't rely on them being named exactly like this", but their presence very much can be relied on. Also, note that it is _Pymoca_ making them and naming them, not RTC-Tools.

You can just get their names via `[x[1] for x in problem_inline.delayed_feedback()]`.

Reply:
I've added a check that the inlined problem constructs fewer constraints than the non-inlined problem.